src/blocks/remote-data-container/components/QueryComponent.tsx

```diff
@@ -0,0 +1,188 @@
+
+	useEffect( () => {
+		onSelectRemoteData( queryInputs );
+	}, [ queryInputs ] );
+
+	function onSelectRemoteData( inputs: RemoteDataQueryInput[] ): void {
+		// if the old queryInputs and new ones are the same, skip this call.
+		if ( JSON.stringify( remoteDataAttribute?.queryInputs ) === JSON.stringify( inputs ) ) {
+			return;
+		}
+
+		void fetch( inputs ).then( () => {
+			if ( innerBlocksPattern ) {
+				insertPatternBlocks( innerBlocksPattern, supportsPagination );
+				return;
+			}
+
+			setShowPatternSelection( true );
+		} );
+	}
+
+	function refreshRemoteData(): void {
+		void fetch( remoteDataAttribute?.queryInputs ?? [ {} ] );
+	}
+
+	function resetPatternSelection(): void {
+		resetInnerBlocks();
+		setShowPatternSelection( false );
+	}
+
+	function resetRemoteData(): void {
+		reset();
+		resetPatternSelection();
+		resetQuery();
+	}
+
+	function onSelectPattern( pattern: BlockPattern ): void {
+		insertPatternBlocks( pattern, supportsPagination );
+		setShowPatternSelection( false );
+	}
+
+	function updateRemoteData( remoteData?: RemoteData ): void {
+		if ( hasRemoteDataChanged( remoteDataAttribute, remoteData ) ) {
+			setAttributes( { remoteData } );
+		}
+	}
+
+	function onUpdateQueryInputs(
+		newSelectorQueryKey: string,
+		inputs: RemoteDataQueryInput[]
+	): void {
+		if ( ! remoteDataAttribute ) {
+			return;
+		}
+
+		updateRemoteData( {
+			...remoteDataAttribute,
+			queryInputs: inputs,
+			// This will always be the display query key.
+			selectorQueryKey: newSelectorQueryKey,
+			displayQueryKey,
+		} );
+		refreshRemoteData();
+	}
+
+	if ( showPatternSelection ) {
+		const supportedPatterns = getSupportedPatterns( data?.results[ 0 ] );
+
+		return (
+			<PatternSelection
+				blockName={ blockName }
+				onCancel={ resetPatternSelection }
+				onSelectPattern={ onSelectPattern }
+				supportedPatterns={ supportedPatterns }
+			/>
+		);
+	}
+
+	return (
+		<>
+			{ ! hasMultiSelection() && data && (
+				<InspectorControls>
+					<OverridesPanel
+						blockConfig={ blockConfig }
+						remoteData={ data }
+						updateRemoteData={ updateRemoteData }
+					/>
+					<DataPanel
+						refreshRemoteData={ refreshRemoteData }
+						remoteData={ data }
+						resetRemoteData={ resetRemoteData }
+					/>
+					<QueryInputsPanel
+						onUpdateQueryInputs={ onUpdateQueryInputs }
+						remoteData={ data }
+						selectors={ getSelectorsForDisplayQuery( blockName, displayQueryKey ) }
+					/>
+				</InspectorControls>
+			) }
+			{ loading && (
+				<div className="remote-data-blocks-loading-overlay">
+					<Spinner
+						style={ {
+							height: '50px',
+							width: '50px',
+						} }
+					/>
+				</div>
+			) }
+			<InnerBlocks />
+		</>
+	);
+}
+
+export function QueryComponent( props: QueryComponentProps ) {
+	const blockProps = useBlockProps( { className: CONTAINER_CLASS_NAME } );
+
+	return (
+		<>
+			<div { ...blockProps }>
+				<EditErrorBoundary blockTitle={ getBlockTitle( props.blockName ) }>
```

Comment:
Is having an ErrorBoundary here, and in the Edit component the right way to handle errors from each component? At the top level, the error from inside QueryComponent wasn't handled, but it was when it was split into two like this. Realistically, does Edit even need an error boundary now? I kept it to be safe but I feel like it could be skipped.